bigbluebutton-html5/imports/ui/components/external-video-player/external-video-player-graphql/component.tsx

```diff
@@ -0,0 +1,360 @@
+  const [mute, setMute] = React.useState(false);
+  const [key, setKey] = React.useState(uniqueId('react-player'));
+  const [volume, setVolume] = React.useState(1);
+  const [subtitlesOn, setSubtitlesOn] = React.useState(false);
+  const [played, setPlayed] = React.useState(0);
+  const [loaded, setLoaded] = React.useState(0);
+  const playerRef = useRef<ReactPlayer>();
+  const playerParentRef = useRef<HTMLDivElement| null>(null);
+  const timeoutRef = useRef<ReturnType<typeof setTimeout>>();
+
+  useEffect(() => {
+    timeoutRef.current = setTimeout(() => {
+      setAutoPlayBlocked(true);
+    }, AUTO_PLAY_BLOCK_DETECTION_TIMEOUT_SECONDS * 1000);
+
+    layoutContextDispatch({
+      type: ACTIONS.SET_HAS_EXTERNAL_VIDEO,
+      value: true,
+    });
+
+    layoutContextDispatch({
+      type: ACTIONS.SET_PRESENTATION_IS_OPEN,
+      value: true,
+    });
+  }, []);
+
+  useEffect(() => {
+    if (playerRef.current && !isPresenter) {
+      playerRef.current.seekTo(currentTime, 'seconds');
+    }
+  }, [currentTime]);
+
+  useEffect(() => {
+    if (playerRef.current) {
+      playerRef.current.seekTo(currentTime, 'seconds');
+    }
+  }, [playerRef.current]);
+
+  const handleOnPlay = () => {
+    setAutoPlayBlocked(false);
+    if (isPresenter) {
+      if (timeoutRef.current) {
+        clearTimeout(timeoutRef.current);
+      }
+      const rate = playerRef.current?.getInternalPlayer()?.getPlaybackRate() as number ?? 1;
+      const currentTime = playerRef.current?.getCurrentTime() ?? 0;
+      sendMessage('play', {
+        rate,
+        time: currentTime,
+      });
+    } else if (!playing && !isPresenter) {
+      playerRef.current?.getInternalPlayer().pauseVideo();
+    }
+  };
+
+  const handleOnPause = () => {
+    if (isPresenter) {
+      const rate = playerRef.current?.getInternalPlayer()?.getPlaybackRate() as number ?? 1;
+      const currentTime = playerRef.current?.getCurrentTime() ?? 0;
+      sendMessage('stop', {
+        rate,
+        time: currentTime,
+      });
+    }
+
+    if (!isPresenter && playing) {
+      playerRef.current?.getInternalPlayer().playVideo();
+    }
+  };
+
+  const isMinimized = width === 0 && height === 0;
+
+  // @ts-ignore accessing lib private property
+  const playerName = playerRef.current && playerRef.current.player
+  // @ts-ignore accessing lib private property
+  && playerRef.current.player.player && playerRef.current.player.player.constructor.name as string;
+  let toolbarStyle = 'hoverToolbar';
+
+  if (deviceInfo.isMobile && !showHoverToolBar) {
+    toolbarStyle = 'dontShowMobileHoverToolbar';
+  }
+
+  if (deviceInfo.isMobile && showHoverToolBar) {
+    toolbarStyle = 'showMobileHoverToolbar';
+  }
+
+  return (
+    <Styled.Container
+      style={{
+        height,
+        width,
+        top,
+        left,
+        right,
+      }}
+      isResizing={isResizing}
+      isMinimized={isMinimized}
+    >
+      <Styled.VideoPlayerWrapper
```

Comment:
missing `data-test` prop is needed for external video automated tests:
```suggestion
      <Styled.VideoPlayerWrapper
        data-test="videoPlayer"
```